rs/compression/src/compression.rs

```diff
@@ -0,0 +1,20 @@
+use std::fs::File;
+use std::io::BufWriter;
+
+use anyhow::Result;
+
+pub trait IntSeqEncoder {
+    /// Creates an encoder
+    fn new_encoder(universe: Option<usize>, size: usize) -> Self
+    where
+        Self: Sized;
```

Comment:
you don't need this. the `new` function should only exist for each individual encoder. that way you don't need that Option (since `NoEncoder` doesn't need the universe)